pkg/log/io/logger.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io
+
+import (
+	"fmt"
+	"log/slog"
+
+	sloglogrus "github.com/samber/slog-logrus/v2"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// JSONFormat represents a json logger
+	JSONFormat string = "json"
+
+	// DebugLevel is the debug level
+	DebugLevel = "debug"
+
+	// InfoLevel is the info level
+	InfoLevel = "info"
+
+	// WarnLevel is the warn level
+	WarnLevel = "warn"
+
+	// ErrorLevel is the error level
+	ErrorLevel = "error"
+)
+
+var (
+	// levelMap transforms a slog.Level to a logrus.Level
+	levelMap = map[slog.Level]logrus.Level{
+		slog.LevelDebug: logrus.DebugLevel,
+		slog.LevelInfo:  logrus.InfoLevel,
+		slog.LevelWarn:  logrus.WarnLevel,
+		slog.LevelError: logrus.ErrorLevel,
+	}
+
+	// DefaultLogLevel is the default log level
+	DefaultLogLevel = slog.LevelWarn
+)
+
+// oktetoLogger is used for recording and categorizing log messages at different levels (e.g., info, debug, warning).
+// These log messages can be filtered based on the log level set by the user.
+// Messages with log levels lower than the user-defined log level will not be displayed to the user.
+type oktetoLogger struct {
+	*slog.Logger
+	slogLeveler *slog.LevelVar
+
+	logrusLogger    *logrus.Logger
+	logrusFormatter logrus.Formatter
+}
+
+// newOktetoLogger returns an initialised oktetoLogger
+func newOktetoLogger() *oktetoLogger {
+	leveler := new(slog.LevelVar)
+	leveler.Set(DefaultLogLevel)
+
+	logrusLogger := logrus.New()
+	logrusLogger.SetLevel(levelMap[DefaultLogLevel])
+	logrusFormatter := &logrus.TextFormatter{}
+	logrusLogger.SetFormatter(logrusFormatter)
+
+	logger := slog.New(sloglogrus.Option{Level: leveler, Logger: logrusLogger}.NewLogrusHandler())
+	return &oktetoLogger{
+		slogLeveler: leveler,
+		Logger:      logger,
+
+		logrusLogger:    logrusLogger,
+		logrusFormatter: logrusFormatter,
+	}
+}
+
+// SetLevel sets the level of the logger
+func (ol *oktetoLogger) SetLevel(lvl string) {
+	slogLevel, err := parseLevel(lvl)
+	if err == nil {
+		ol.slogLeveler.Set(slogLevel)
+		ol.logrusLogger.SetLevel(levelMap[slogLevel])
```

Comment:
why both?

Reply:
`logrusLogger` needs to set it's own level if we want to keep the retrocompatibility. It will be removed when we migrate the whole logger but for now we need to have that

Reply:
why do we need backward compatibility? in which case? i don't understand why after the migration we don't need that backward compatibility anymore, can you clarify it?

Reply:
can we add TODO comments to all this code that needs to change once full migration is done?